src/data/dataset.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python3
+"""A dataset class lyrics and MIDI data for the few-shot-music-gen project
+"""
+import os
+import logging
+import time
+
+import numpy as np
+
+
+log = logging.getLogger('few-shot')
+
+
+class Dataset(object):
+    """A class for train/val/test sets.
+
+    This class is initialized with the following arguments:
+    Arguments:
+        root (str): the root directory of the dataset
+        split ("train", "val", or "test"): the split of the dataset which
+            this object represents.
+        loader (LyricsLoader or MIDILoader): the object used for reading and
+            parsing
+        split_proportions (tuple of three numbers): the unnormalized
+            (train, val, test) split.
+        persist (bool): persists the train/val/test split information in csv
+            files in `root`, so future runs will use the same splits. If those
+            csvs already exist, the sampler uses the splits from those files.
+        cache (bool): if true, caches the loaded/parsed songs in memory.
+            Otherwise it loads and parses songs on every episode.
+        min_songs (int): the minimum number of songs which an artist must have.
+            If they don't have `min_songs` songs, they will not be present in
+            the dataset.
+    """
+    def __init__(self, root, split, loader, split_proportions=(8,1,1),
+            persist=True, cache=True, min_songs=0):
+        self.root = root
+        self.cache = cache
+        self.cache_data = {}
+        self.loader = loader
+        split_csv_path = os.path.join(root, '%s.csv' % split)
+        if persist and os.path.exists(split_csv_path):
+            split_csv = open(split_csv_path, 'r')
+            self.artists = [line.strip() for line in split_csv.readlines()]
+            split_csv.close()
+        else:
+            dirs = []
+            for artist in os.listdir(root):
+                if os.path.isdir(os.path.join(root, artist)):
+                    dirs.append(artist)
+            artists = []
+            skipped_count = 0
+            num_dirs = len(dirs)
+            last_log = 0
+            for artist_index, artist in enumerate(dirs):
+                # log progress every second
+                if time.time() - last_log >= 1:
+                    log.info("Preprocessing data. %s%%" % int(100*artist_index/num_dirs))
+                    last_log = time.time()
+                songs = os.listdir(os.path.join(root, artist))
+                songs = [song for song in songs if not os.path.isdir(song)]
```

Comment:
The check `is not os.path.isdir(song)` is not needed.

Reply:
:+1: